test/tuned_models.jl

```diff
@@ -341,4 +341,48 @@ end
 
 end
 
+@testset_accelerated "weights and class_weights are being passed" accel begin
+    # we'll be tuning using 50/50 holdout
+    X = (x=fill(1.0, 6),)
+    y = coerce(["a", "a", "b", "a", "a", "b"], OrderedFactor)
+    w = [1.0, 1.0, 100.0, 1.0, 1.0, 100.0]
+    class_w = Dict("a" => 2.0, "b" => 100.0)
+
+    model = DecisionTreeClassifier()
+
+    # the first supports weights, the second class weights:
+    ms=[MisclassificationRate(), MulticlassFScore()]
+
+    resampling=Holdout(fraction_train=0.5)
+
+    # without weights:
+    tmodel = TunedModel(
+        resampling=resampling,
+        models=fill(model, 5),
+        measures=ms,
+        acceleration=accel
+    )
+    mach = machine(tmodel, X, y)
+    fit!(mach, verbosity=0)
+    measurement = report(mach).best_history_entry.measurement
+    e = evaluate(model, X, y, measures=ms, resampling=resampling)
+    @test measurement == e.measurement
+
+    # with weights:
+    tmodel.weights = w
+    tmodel.class_weights = class_w
+    fit!(mach, verbosity=0)
+    measurement_weighted = report(mach).best_history_entry.measurement
+    e_weighted = evaluate(model, X, y;
+                          measures=ms,
+                          resampling=resampling,
+                          weights=w,
+                          class_weights=class_w,
+                          verbosity=-1)
+    @test measurement_weighted == e_weighted.measurement
+
+    # check both measures are different when they are weighted:
+    @test !any(measurement .== measurement_weighted)
```

Comment:
Is there a way to test if the passed class_w was in-fact used here?

`levels(keys(class_w))` is `[a, b]`, while the resultant is in `[b, a]` order.

On `dump`ing `evaluate`, I see that FScore returns `LittleDict`, but not sure how that can be accessed. `measurement` is only a vector.
```
 PerformanceEvaluation{Vector{MLJBase.Aggregated}, Vector{Float64}, Vector{typeof(predict_mode)}, Vector{Vector{Float64}}, Vector{Missing}, Vector{NamedTuple{(:tree_or_leaf, :encoding), Tuple{DecisionTree.Leaf{UInt32}, Dict{CategoricalArrays.CategoricalValue{String, UInt32}, UInt32}}}}, Vector{NamedTuple{(:classes_seen, :print_tree), Tuple{CategoricalArrays.CategoricalVector{String, UInt32, String, CategoricalArrays.CategoricalValue{String, UInt32}, Union{}}, Main.Models.TreePrinter{DecisionTree.Leaf{UInt32}}}}}}
  measure: Array{MLJBase.Aggregated}((2,))
    1: MisclassificationRate MisclassificationRate()
    2: MulticlassFScore{Float64, MLJBase.MacroAvg, LittleDict}
  measurement: Array{Float64}((2,)) [33.333333333333336, 0.8]
:
:
```

Reply:
I believe such test would have be done at MLJBase.jl since MLJTuning creates `Resampler`s which are implemented at MLJTuning.jl

Reply:
According to https://github.com/JuliaAI/MLJBase.jl/blob/dev/src/resampling.jl#L1194-L1202 and https://github.com/JuliaAI/MLJBase.jl/blob/dev/src/resampling.jl#L1146-L1156 `class_weights` are not always used.

Reply:
> MLJBase doesn't expose such interface.

Ok

>it actually returns a vector

Actually it dispatches on `f.return_type`, and as it's `LittleDict` that would be this [block](https://github.com/JuliaAI/MLJBase.jl/blob/dev/src/measures/finite.jl#L1187-1191)

Reply:
@ven-k  No I can't really think of a more direct test\, but I think the test I give is sufficient evidence that class weights are being passed. Are you not convinced?

@OkonSamuel I'm not sure I'm understanding the issue around vector vs dict here. Are you satisfied now?